### ISSUE TITLE
UWP game on XBox receives mysterious InputKey events

### ISSUE THREAD

**Issue by nunodonato** [godotengine/godot#9214]
**Operating system or device - Godot version:** UWP(XBox), Godot 2.2

**Issue description:**
Although I'm just using the xbox controller, my code is printing InputKey events everytime I press any button or move the sticks in the controller.
The scancode is: 33554431, which seems to be a very strange code :)

**Comment by vnen** [comment]
Yeah, the Gamepad emit input key events, I don't know if there's anyway around it.

**Comment by nunodonato** [comment]
I can force ignore that specific scancode.. but doesnt sound like a very elegant solution

**Comment by punto-** [comment]
It's a specific scancode? Maybe it's a special "xbox key"?

**Comment by nunodonato** [comment]
@punto- yeah, I always get scancode 33554431 no matter which button I press.

**Comment by akien-mga** [comment]
That's `KEY_UNKNOWN`:
> KEY_UNKNOWN = 33554431

**Comment by akien-mga** [comment]
Maybe some other VK constant to convert to the relevant Godot scancode? https://github.com/godotengine/godot/blob/master/platform/windows/key_mapping_win.cpp

**Comment by akien-mga** [comment]
@nunodonato Maybe add a print statement to get the `p_code` value in https://github.com/godotengine/godot/blob/master/platform/windows/key_mapping_win.cpp#L240 ?

**Comment by nunodonato** [comment]
@akien-mga will do. I need the devkit for this, so it will take a while :/

**Comment by vnen** [comment]
There are specific VirtualKey codes for gamepad buttons. Check https://docs.microsoft.com/en-us/uwp/api/windows.system.virtualkey. The page don't show the numbers but the range is 0xC3-0xDA.

I don't really know how to deal with them. Should they just get ignored or should we add Gamepad keycodes to Godot?

**Comment by akien-mga** [comment]
CC @Hinsbart :)

**Comment by akien-mga** [comment]
We have now entered release freeze for Godot 3.0 and want to focus only on release critical issues for that milestone. Therefore, we're moving this issue to the 3.1 milestone, though a fix may be made available for a 3.0.x maintenance release after it has been tested in the master branch during 3.1 development. If you consider that this issue is critical enough to warrant blocking the 3.0 release until fixed, please comment so that we can assess it more in-depth.

**Comment by Hinsbart** [comment]
> I don't really know how to deal with them. Should they just get ignored or should we add Gamepad keycodes to Godot?

Since we have proper gamepad support I think it's best to just ignore those keyboard events.

**Comment by reduz** [comment]
Assigned issue to you guys, we should do something with it or close it I guess?

**Comment by GeorgeS2019** [comment]
can someone assign this to milestone 4.0 just like [this](https://github.com/godotengine/godot/issues/20271) and [this](https://github.com/godotengine/godot/issues/19320)?

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?
